src/utils.c

```diff
@@ -80,6 +80,52 @@ char *str_concat_many(int count, ...)
     return combined;
 }
 
+char *str_replace(char *search, char *replace, char *subject) {
+    char *result;       // the return string
+    char *ins;          // the next insert point
+    char *tmp;          // varies
+    size_t len_search;  // length of search (the string to remove)
+    size_t len_replace; // length of replace (the string to replace search with)
+    size_t len_front;   // distance between search and end of last replace
+    int count;          // number of replacements
+
+    // sanity checks and initialization
+    if (!subject || !search)
+        return NULL;
+    len_search = strlen(search);
+    if (len_search == 0)
+        return NULL; // empty search causes infinite loop during count
+    if (!replace)
+        replace = "";
+    len_replace = strlen(replace);
+
+    // count the number of replacements needed
+    ins = subject;
+    for (count = 0; tmp = strstr(ins, search); ++count) {
```

Comment:
does this break when tmp == NULL?

Reply:
```
utils.c:104:25: warning: using the result of an assignment as a condition
      without parentheses [-Wparentheses]
    for (count = 0; tmp = strstr(ins, search); ++count) {
                    ~~~~^~~~~~~~~~~~~~~~~~~~~
utils.c:104:25: note: place parentheses around the assignment to silence this
      warning
    for (count = 0; tmp = strstr(ins, search); ++count) {
                        ^
                    (                        )
utils.c:104:25: note: use '==' to turn this assignment into an equality
      comparison
    for (count = 0; tmp = strstr(ins, search); ++count) {
                        ^
                        ==
1 warning generated.

```

Reply:
Yes. It breaks on `tmp == NULL`. I can rework this line to avoid the compiler warning.